index.js

```diff
@@ -17,16 +17,11 @@ function transformer(ast) {
   visit(ast, 'heading', visitor);
 
   function visitor(node) {
-    var id = slugs.slug(toString(node));
-
-    if (!node.data) {
-      node.data = {};
-    }
-
-    if (!node.data.hProperties) {
-      node.data.hProperties = {};
-    }
+    node.data = node.data || {};
+    node.data.hProperties = node.data.hProperties || {};
 
+    const value = node.data.hProperties.id || toString(node);
+    const id = slugs.slug(value);
```

Comment:
Could you use `var` here? And keep it at the top? I also think it could all be on one line maybe?

Reply:
Could definitely use `var` if that is preferred (should have kept existing convention) and make one line.  We cannot move to the top as the `const value` line requires that `node.data.hProperties` be defined.

Reply:
Ah, you’re right for the last one. OK! 👍